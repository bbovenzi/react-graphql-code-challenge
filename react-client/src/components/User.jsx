import React from "react";

import style from 'styles/User.module.css';
import Link from 'components/Link';

function User({ user }) {
  return (
    <div className={style.user}>
      <h2 className={style.name}>{user.name}</h2>
      <Link href={`mailto:${user.email}`}>
        <p>{user.email}</p>
      </Link>
      <Link href={`tel:${user.phone}`}>
        <p>{user.phone}</p>
      </Link>
    </div>
  )
}

export default User;
