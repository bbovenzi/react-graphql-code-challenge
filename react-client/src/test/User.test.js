import React from 'react';
import renderer from 'react-test-renderer';
import User from 'components/User';

test('User matches snapshot', () => {
  const mockUser = {
    id: 1,
    name: 'name',
    email: 'email@email.com',
    phone: '5555555555',
  }
  const component = renderer.create(
    <User user={mockUser} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
