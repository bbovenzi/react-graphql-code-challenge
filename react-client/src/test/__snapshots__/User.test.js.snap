// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches snapshot 1`] = `
<div
  className="user"
>
  <h2
    className="name"
  >
    name
  </h2>
  <a
    className="link"
    href="mailto:email@email.com"
    rel="noopener"
    target="_blank"
  >
    <p>
      email@email.com
    </p>
  </a>
  <a
    className="link"
    href="tel:5555555555"
    rel="noopener"
    target="_blank"
  >
    <p>
      5555555555
    </p>
  </a>
</div>
`;
